src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -2149,6 +2159,111 @@ static void assertBindIdentifierTargets(InMethodBinder? inMethodBinder, Concurre
 #endif
         }
 
+        /// <summary>
+        /// Report a diagnostic for any 'field' or 'value' identifier in the bound tree where the
+        /// meaning will change when the identifier is considered a contextual keyword.
+        /// </summary>
+        private static void ReportFieldOrValueContextualKeywordConflicts(MethodSymbol method, BoundNode node, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(method.MethodKind is MethodKind.PropertyGet or MethodKind.PropertySet);
+            Debug.Assert(method.AssociatedSymbol is PropertySymbol);
+
+            PooledDictionary<SyntaxNode, SyntaxToken>? valueIdentifiers = null;
+
+            foreach (var token in node.Syntax.DescendantTokens())
+            {
+                Debug.Assert(token.Parent is { });
+                if (token.Kind() != SyntaxKind.IdentifierToken)
+                {
+                    continue;
+                }
+                var syntax = token.Parent;
+                switch (syntax)
```

Comment:
Why check the parent syntax kinds at all? Is there any IdentifierToken we don't want to flag?

Reply:
> Why check the parent syntax kinds at all? Is there any IdentifierToken we don't want to verify/flag?

I think we want to verify all `IdentifierToken` cases so we probably don't need to check `token.Parent`. But there aren't that many distinct cases, so I thought it better to check explicitly, at the risk of missing a case, rather than pulling in other cases unintentionally. And as you mentioned elsewhere, we shouldn't need to update this list when new syntax nodes are added in later language versions.